src/process/main.cpp

```diff
@@ -87,6 +89,50 @@ int main( int argc, char *argv[] )
   }
   QgsApplication::setPrefixPath( myPrefixPath, true );
 
+  // Set translation context for qgis_process: necessary for naming of additional fields created by algorithms
+  // The command line parameter evaluation is done in qgisprocessing.cpp,
+  // which is too late for setting the language context.
+
+  QString translationCode;
+  int i = 1;
+  for(; i < argc; ++i)
+    if ( i < argc && ( argv[i] == QLatin1String( "--lang" ) || argv[i] == QLatin1String( "-l" ) ) )
+    {
+        translationCode = argv[++i];
+    }
+
+  QString myUserTranslation;
+  myUserTranslation = QLocale::system().name();                   // e.g. "de_DE"
+  myUserTranslation.truncate(myUserTranslation.lastIndexOf('_')); // e.g. "de"
+
+  //
+  // Priority of translation context is:
+  //  - command line:  from --lang xx or -l xx
+  //  - system locale: from QLocale::system().name()
+  //
+
+  if ( translationCode.isNull() || translationCode.isEmpty())
+  {
+    translationCode = myUserTranslation;
+  }
+  QgsApplication::setTranslation( translationCode );
+
+  if ( translationCode != QLatin1String( "C" ) )
+  {
+    QTranslator *mQgisTranslator = nullptr;
```

Comment:
"m" prefix is reserved for member variables

```suggestion
    QTranslator *qgisTranslator = nullptr;
```